### PR TITLE
Add asio standalone example

### PR DESCRIPTION
**Pull request by kingsamchen**
Add Example for the using asio in standalone mode

### CODE REVIEW

File: examples/asio-standalone/main.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    asio::signal_set signals(ioc, SIGINT, SIGTERM);
+    signals.async_wait([&ioc](std::error_code, int) {
+        ioc.stop();
+    });
+
+    ioc.run();
+
+    return 0;
+}
```
[the remark this answers is not in the document]
Comment (kingsamchen) [reply examples/asio-standalone/main.cpp]: I changed the example to this [asynchronous tcp echo server](https://think-async.com/Asio/asio-1.16.1/src/examples/cpp11/echo/async_tcp_echo_server.cpp), which is more instructive. Thanks for the suggestion.

File: examples/asio-standalone/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+include(../../cmake/CPM.cmake)
+
+find_package(Threads REQUIRED)
+
+CPMAddPackage(
+  NAME asio
+  GITHUB_REPOSITORY chriskohlhoff/asio
+  GIT_TAG asio-1-16-1   # cannot use VERSION
+)
```
[the remark this answers is not in the document]
Comment (kingsamchen) [reply examples/asio-standalone/CMakeLists.txt]: The `VERSION` param is added, and it now looks neat XD.

Also thanks for the using-URL tip, though I would prefer using version tag, it is more intuitive, and consistent with most usages, IMO; and for those who use global cache, the repository would be shallow-cloned, also saving much bandwidth.

Nonetheless, it still would be a great idea to add it as a supplement case in wiki page.

**Comment by kingsamchen** [comment]
@TheLartians Hi, sorry for the interruption, is there any issue still need to be resolved with the latest commit ? would be great on receiving your feedback!

File: examples/asio-standalone/main.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
+
```
Comment (TheLartians) [inline examples/asio-standalone/main.cpp:1]: It seems you've removed the copyright and licence notice from original author in the top of the example. However, the [boost software licence](https://www.boost.org/LICENSE_1_0.txt) requires the Licence to be left in all derived works.

```suggestion
//
// async_tcp_echo_server.cpp
// ~~~~~~~~~~~~~~~~~~~~~~~~~
//
// Copyright (c) 2003-2020 Christopher M. Kohlhoff (chris at kohlhoff dot com)
//
// Distributed under the Boost Software License, Version 1.0.
// (See http://www.boost.org/LICENSE_1_0.txt)
//

```

**Pull request merged**